### ISSUE TITLE
self[] & static[] type hints return the local classes auto completions.

### ISSUE THREAD

**Issue by brad-jones** [php-integrator/atom-autocompletion#85]
Hopefully this example illustrates the issue:

```php
class Foo
{
    /**
     * @return self[]|static[]
     */
    public static function bar()
    {
        return [ new Foo, new Foo, new Foo, etc... ];
    }
}

class Baz
{
    public $abc;
    public $xyz;
    
    public function __construct()
    {
        Foo::bar()[0]-> // this does not autocomplete at all.
        
        foreach (Foo::bar() as $foo)
        {
            $foo-> // this returns Baz's completions (ie: $abc & $xyz)
        }
    }
}
```

**Comment by Gert-dev** [comment]
Looks like you've stumbled upon a corner case that isn't working properly in the core. Marking as bug.

**Comment by brad-jones** [comment]
Just tested this and still failing with v2.1 of the core.

**Comment by Gert-dev** [comment]
Is there a specific item that is still failing and I missed? It was definitely failing, but I can get it to work properly now:

![screenshot](https://cloud.githubusercontent.com/assets/6823929/21453452/78a08a8c-c90f-11e6-89f4-cf4c5716b1e9.png)

![screenshot 2](https://cloud.githubusercontent.com/assets/6823929/21453458/8b969fdc-c90f-11e6-94f3-21bac1289082.png)

**Comment by brad-jones** [comment]
Sorry for the delay in replying, the new year and all...

Anyway just did some quick testing on this and it would seem to only be working for a return type of self[] and not static[].

This Works:
```php
/**
 * @return self[]
 */
```

This Fails:
```php
/**
 * @return static[]
 */
```

**Comment by Gert-dev** [comment]
Thanks, I can reproduce it now. I've released base 2.1.4 that pulls in the new core that contains the fix.
